### PR TITLE
feat(components): [date-picker] add align option

### LINKED ISSUE #6406: [Feature Request] [Enhancement] [date-picker] Implement the "align" option for the date picker component as in element-ui
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `date-picker`

## Description

The element-ui date-picker allowed **align** to be set to `left/center/right` and defaulted to `left`. It doesn't seem like the option is available in element-plus and only aligns as `center`.

```vue
<template>
  <el-date-picker
    v-model="dateRange"
    type="daterange"
    :size="size"
    :clearable="false"
    range-separator="-"
    align="start"
    start-placeholder="Start date"
    end-placeholder="End date"
    placeholder="Select date range"
    :shortcuts="datePickerShortcuts"
    @change="setSelectedRange"
    format="DD MMM YYYY"
  />
</template>
```

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by nieyuyao**
Closes #6406 

- [x] Adds `align` option to the `ElDatePicker` and `ElTimePicker`

----

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by JeremyWuuuuu** [comment]
@nieyuyao First of all, thank you for contributing to Element Plus.

One thing I do want to mention here is that the API seems irrelevant with `align` instead it relates to the popup content, so I think we should give the API a `namespace` to avoid misunderstanding here.

File: packages/components/time-picker/src/common/picker.vue [hunk 1]
```diff
@@ -654,6 +663,25 @@ export default defineComponent({
       props,
     })
 
+    const getRealAlign = (propAlign: string) => {
+      return (
+        ['start', 'middle', 'end'].find((align) => align === propAlign) ??
+        'start'
+      )
+    }
```
Comment (JeremyWuuuuu) [inline packages/components/time-picker/src/common/picker.vue:671]: Why not make this a `computed`?

```ts
const _alignment = computed(() => {
  return ['start', 'middle', 'end'].find((align) => align === props.align) ?? 'start'
})
```

**Comment by nieyuyao** [comment]
> @nieyuyao First of all, thank you for contributing to Element Plus.
> 
> One thing I do want to mention here is that the API seems irrelevant with `align` instead it relates to the popup content, so I think we should give the API a `namespace` to avoid misunderstanding here.

Okay, I see. For example, we name it `pickerContentAlign`? 🤔

**Comment by JeremyWuuuuu** [comment]
> Okay, I see. For example, we name it pickerContentAlign? 🤔

Yes! Something like that, the main idea is that we do not want to make our user confused with so many APIs. It is always a better option to add a namespace

**Comment by JeremyWuuuuu** [comment]
@nieyuyao Just checking in to see if I can help you with anything, let me know if you do.

File: packages/components/time-picker/src/common/picker.vue (hunk 1 above)
Comment (nieyuyao) [reply packages/components/time-picker/src/common/picker.vue]: done

**Comment by xiaoxian521** [comment]
Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

**Pull request closed without merging**